### PR TITLE
Use 'CARGO_WORKSPACE_DIR' if present, closes #33

### LINKED ISSUE #33: `to_abs_ws_path` goes too far up
Running RA tests from within a `rust-lang/rust` checkout fails, because `expect-test` climbs all the way up to `rust/Cargo.toml` instead of being happy with `rust/src/tools/rust-analyzer/Cargo.toml`, and then the paths are all wrong.

See https://github.com/rust-lang/rust/pull/99444#issuecomment-1188844202 for real-world details.

### PR DESCRIPTION
**Pull request by fasterthanlime**
Will test that out on https://github.com/rust-lang/rust/pull/99444 before taking it out of draft.

### CODE REVIEW

**Comment by fasterthanlime** [comment]
This works well enough to resolve the problems we saw in https://github.com/rust-lang/rust/pull/99444

using this PR, when adding an env section to rust-analyzer's .cargo/config.toml, all tests pass from within the rust repo

**Review by Veykril — APPROVED** [review]
Can you bump the crate version in the cargo.toml? CI will auto-publish the version when we merge it then.

**Comment by fasterthanlime** [comment]
> Can you bump the crate version in the Cargo.toml? CI will auto-publish the version when we merge it then. And a small CHANGELOG.md entry would be nice as well :)

Done!

File: src/lib.rs [hunk 1]
```diff
@@ -591,11 +591,7 @@ fn lit_kind_for_patch(patch: &str) -> StrLitKind {
     let has_dquote = patch.chars().any(|c| c == '"');
     if !has_dquote {
         let has_bslash_or_newline = patch.chars().any(|c| matches!(c, '\\' | '\n'));
-        return if has_bslash_or_newline {
-            StrLitKind::Raw(1)
-        } else {
-            StrLitKind::Normal
-        };
+        return if has_bslash_or_newline { StrLitKind::Raw(1) } else { StrLitKind::Normal };
```
Comment (fasterthanlime) [inline src/lib.rs:594]: crap, "format on save" strikes again

**Pull request merged**